src/app/external-cluster-wizard/steps/external-cluster/provider/gke/component.ts

```diff
@@ -345,20 +352,10 @@ export class GKEClusterSettingsComponent
       });
   }
 
-  private _getGKEKubernetesVersions(): void {
+  private _getGKEKubernetesVersions(zone: string, mode: string): void {
     this.kubernetesVersions = [];
-    let zone = this.controlValue(Controls.Zone);
-    let mode = this.controlValue(Controls.KubernetesVersionMode);
     let releaseChannel: string;
 
-    if (!zone) {
```

Comment:
so now if you clear zone and try to change release channel you will get the error notification that zone can't be empty did you did it on purpose ? 
![Screenshot from 2022-09-14 18-55-34](https://user-images.githubusercontent.com/85109141/190203654-a60a69c0-146c-4ab0-9b55-44e5a0a6c142.png)

Reply:
- I have kept `if (zone && mode) {` because we can send API call by sending default value from code event notifying what value was used for Zone so until user won't select zone we will not trigger API to get kuberneters version. As due to update of Zone to combobox it opens the opportunity to clear zone value which we can't restrict user but we can handle UI form to stop moving forward.

~Secondly, it always a good approach to pass as parameters IMO 

- lastly, calling setValue inside triggers formValue changes which was again calling method hence twice method was called not the API once both value was set than it was invoking API